src/Lykke.Job.BlockchainOperationsExecutor/Workflow/CommandHandlers/OperationExecution/GenerateActiveTransactionIdCommandsHandler.cs

```diff
@@ -1,25 +1,109 @@
 ﻿using System;
 using System.Threading.Tasks;
+using Common.Log;
 using JetBrains.Annotations;
+using Lykke.Common.Log;
 using Lykke.Cqrs;
 using Lykke.Job.BlockchainOperationsExecutor.Core.Domain.OperationExecutions;
 using Lykke.Job.BlockchainOperationsExecutor.Workflow.Commands.OperationExecution;
 using Lykke.Job.BlockchainOperationsExecutor.Workflow.Events.OperationExecution;
+using Lykke.Job.BlockchainOperationsExecutor.Workflow.Events.TransactionExecution;
 
 namespace Lykke.Job.BlockchainOperationsExecutor.Workflow.CommandHandlers.OperationExecution
 {
     public class GenerateActiveTransactionIdCommandsHandler
     {
         private readonly IActiveTransactionsRepository _activeTransactionsRepository;
+        private readonly IOperationExecutionsRepository _repository;
+        private readonly RetryDelayProvider _retryDelayProvider;
+        private readonly ILog _log;
 
-        public GenerateActiveTransactionIdCommandsHandler(IActiveTransactionsRepository activeTransactionsRepository)
+        public GenerateActiveTransactionIdCommandsHandler(IActiveTransactionsRepository activeTransactionsRepository, 
+            IOperationExecutionsRepository repository,
+            RetryDelayProvider retryDelayProvider,
+            ILogFactory logFactory)
         {
             _activeTransactionsRepository = activeTransactionsRepository;
+            _repository = repository;
+            _retryDelayProvider = retryDelayProvider;
+            _log = logFactory.CreateLog(this);
         }
 
         [UsedImplicitly]
         public async Task<CommandHandlingResult> Handle(GenerateActiveTransactionIdCommand command, IEventPublisher publisher)
         {
+            if (command.IsCashout 
+                && command.ActiveTransactioNumber > 0)  // generate active transaction id after rebuild command
+            {
+                var aggregate = await _repository.GetAsync(command.OperationId);
+
+                var loggingContext = new
+                {
+                    aggregate.OperationId,
+                    aggregateActiveTransactionNumber = aggregate.ActiveTransactionNumber,
+                    commandActiveTransactionNumber = command.ActiveTransactioNumber
+                };
+
+                if (aggregate.ActiveTransactionNumber > command.ActiveTransactioNumber)
+                {
+                    _log.Warning("Command already handled" +
+                                 "Do nothing",
```

Comment:
Looks like it should be single line with a ".  " between "handled" and "Do nothing". Or at least ". " in the end of the first line, as it made below.